lib/Castle/Request.php

```diff
@@ -75,26 +75,14 @@ public static function getUserAgent()
     return null;
   }
 
-  public static function getCID()
+  public static function getClientId()
   {
-    $cid = Castle::getCookieStore()->read('__cid');
-
-    // If the cookie is specifically not defined or not set in the cookie jar
-    // we'll return the special value '?'. This doesn't have any effect on
-    // functionality.
-    if (empty($cid)) {
-      $cid = '?';
+    if (array_key_exists('HTTP_X_CASTLE_DEVICE_ID', $_SERVER)) {
+      return $_SERVER['HTTP_X_CASTLE_DEVICE_ID'];
+    } else {
+      $cid = Castle::getCookieStore()->read('__cid');
+      return empty($cid) ? '' : $cid;
```

Comment:
This logic need to be kept in order for cURL to not strip the header. See previous PR by @nullaus